core/git.schema.go

```diff
@@ -0,0 +1,146 @@
+package core
+
+import (
+	"fmt"
+
+	"github.com/graphql-go/graphql"
+	"github.com/moby/buildkit/client/llb"
+	"go.dagger.io/dagger/router"
+)
+
+var _ router.ExecutableSchema = &gitSchema{}
+
+type gitSchema struct {
+	*baseSchema
+}
+
+func (s *gitSchema) Name() string {
+	return "git"
+}
+
+func (s *gitSchema) Schema() string {
+	return `
+	extend type Query {
+		"Query a git repository"
+		git(url: String!): GitRepository!
+	}
+	
+	"A git repository"
+	type GitRepository {
+		"List of branches on the repository"
+		branches: [String!]!
+		"Details on one branch"
+		branch(name: String!): GitRef!
+		"List of tags on the repository"
+		tags: [String!]!
+		"Details on one tag"
+		tag(name: String!): GitRef!
```

Comment:
Tried it out and noticed that `branch` and `tag` are actually interchangeable (you can specify a tag in `branch` or a `branch` in tag and everything works), so maybe it makes more sense to just have a `ref(name: String!)` field for now? And then we can add separate fields in the future that actually enforce the ref being a branch/tag if we want.

Reply:
I have a slight preference for keeping it this way, so that 1) developers can express their intent, and 2) we don't break everyone later. The downside is that some devs may accidentally use tag to get a ref, or vice-versa, and their code will break later when we enforce properly. But I think that will be less annoying than trying to get everyone to stop using `ref`.

I can live with both btw, this is my 5mn bikeshedding break.

Reply:
Yeah my original comment is very much bikeshedding too, I see what you're saying though. Probably a situation where it's not clear what the right answer is until someone tries using it, so we should just merge it as is.